src/Storages/StorageReplicatedMergeTree.cpp

```
@@ -4449,119 +4459,16 @@ void StorageReplicatedMergeTree::exportMergeTreePartitionStatusHandlingTask()
     }
 }
 
-std::vector<ReplicatedPartitionExportInfo> StorageReplicatedMergeTree::getPartitionExportsInfo() const
+std::vector<ReplicatedPartitionExportInfo> StorageReplicatedMergeTree::getPartitionExportsInfo(bool prefer_remote_information) const
 {
-    std::vector<ReplicatedPartitionExportInfo> infos;
-
-    const auto zk = getZooKeeper();
-    const auto exports_path = fs::path(zookeeper_path) / "exports";
-    std::vector<std::string> children;
-    if (Coordination::Error::ZOK != zk->tryGetChildren(exports_path, children))
+    if (prefer_remote_information && getZooKeeper()->isFeatureEnabled(DB::KeeperFeatureFlag::MULTI_READ))
```

This is silently falling back to the local information that does not require ZooKeeper, but maybe we should not do it and simply throw an exception telling the user to use the other option?